docs/proposals/observability.md

```diff
@@ -0,0 +1,315 @@
+- [`otel_span_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_name)
+- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr)
+
+Tracing will be disabled by default. The Application Developer will be able to use this Policy to enable and configure tracing for their routes. This Policy will only be applied if the OTel endpoint has been set by the Cluster Operator on the [Gateway Settings](gateway-settings.md).
+
+Ratio and parent-based tracing should be supported as shown in the [nginx-otel examples](https://github.com/nginxinc/nginx-otel?tab=readme-ov-file#examples).
+
+In the future, this config will be extended to support other functionality, such as those defined in the [NGINX Extensions Proposal](nginx-extensions.md#observability).
+
+## API, Customer Driven Interfaces, and User Experience
+
+The `ObservabilityPolicy` API is a CRD that is a part of the `gateway.nginx.org` Group. It is a namespaced resource that will reference an HTTPRoute as its target.
+
+### Go
+
+Below is the Golang API for the `ObservabilityPolicy` API:
+
+```go
+package v1alpha1
+
+import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type ObservabilityPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the ObservabilityPolicy.
+    Spec ObservabilityPolicySpec `json:"spec"`
+
+    // Status defines the state of the ObservabilityPolicy.
+    Status gatewayv1alpha2.PolicyStatus `json:"status,omitempty"`
+}
+
+type ObservabilityPolicySpec struct {
+    // TargetRef identifies an API object to apply the policy to.
+    // Object must be in the same namespace as the policy.
+    // Support: HTTPRoute
+    TargetRef gatewayv1alpha2.PolicyTargetReference `json:"targetRef"`
+
+    // Tracing allows for enabling and configuring tracing.
+    //
+    // +optional
+    Tracing *Tracing `json:"tracing,omitempty"`
+}
+
+// Tracing allows for enabling and configuring OpenTelemetry tracing.
+type Tracing struct {
+    // Ratio is the percentage of traffic that should be sampled. Integer from 0 to 100.
+    // By default, 100% of http requests are traced. Not applicable for parent-based tracing.
+    //
+    // +optional
+    Ratio *int32 `json:"ratio,omitempty"`
+
+    // Context specifies how to propagate traceparent/tracestate headers. By default is 'ignore'.
+    //
+    // +optional
+    Context *TraceContext `json:"context,omitempty"`
+
+    // SpanName defines the name of the Otel span. By default is the name of the location for a request.
+    //
+    // +optional
+    SpanName *string `json:"spanName,omitempty"`
+
+    // SpanAttributes are custom key/value attributes that are added to each span.
+    //
+    // +optional
+    SpanAttributes map[string]string `json:"spanAttributes,omitempty"`
```

Comment:
Should we mention here that these attributes are merged with the attributes defined in the GatewaySettings?